File: Capacitaciones-Front/src/services/usuarioService.js
import api from './http'

export const usuarioService = {
  buscarTodosUsuarios: () => api.get('/usuario'),
  save: (usuario) => api.post('/usuario', usuario),
  obtenerUsuariosAsignados: (idCapacitacion) =>
    api.get(`/usuario/usuariosPorCapacitacion/${idCapacitacion}`),
  eliminar: (id) => api.delete(`/usuario/${id}`),

  buscarUsuario: (idUsuario) => api.get(`/usuario/usuarioPorId/${idUsuario}`),
}

export default usuarioService

File: Capacitaciones-Back/capacitaciones/src/main/java/com/example/capacitaciones/service/UsuarioService.java
package com.example.capacitaciones.service;

import java.util.List;

import com.example.capacitaciones.DTOs.UsuarioDTO;
import com.example.capacitaciones.model.Usuario;

public interface UsuarioService {
    UsuarioDTO save(Usuario usuario);
    List<UsuarioDTO> findAll();
    void delete(Long idUsuario);
    Usuario findByEmail(String email);
    List<UsuarioDTO> findAllByCapacitacion(Long idCapacitacion);
    Usuario findById(Long idUsuario);
}
